pkg/usagestats/stats.go

```diff
@@ -0,0 +1,65 @@
+package usagestats
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"runtime"
+	"time"
+
+	jsoniter "github.com/json-iterator/go"
+	"github.com/prometheus/common/version"
+)
+
+var (
+	httpClient    = http.Client{Timeout: 5 * time.Second}
+	usageStatsURL = "https://stats.grafana.org/agent-usage-report"
+)
+
+// Report is the payload to be sent to stats.grafana.org
+type Report struct {
+	UsageStatsID string                 `json:"usageStatsId"`
+	CreatedAt    time.Time              `json:"createdAt"`
+	Interval     time.Time              `json:"interval"`
+	Version      string                 `json:"version"`
+	Metrics      map[string]interface{} `json:"metrics"`
+	Os           string                 `json:"os"`
+	Arch         string                 `json:"arch"`
+}
+
+func sendReport(ctx context.Context, seed *AgentSeed, interval time.Time, metrics map[string]interface{}) error {
+	report := Report{
+		UsageStatsID: seed.UID,
+		CreatedAt:    seed.CreatedAt,
+		Version:      version.Print("agent"),
+		Os:           runtime.GOOS,
+		Arch:         runtime.GOARCH,
+		Interval:     interval,
+		Metrics:      metrics,
+	}
+	out, err := jsoniter.MarshalIndent(report, "", " ")
+	if err != nil {
+		return err
+	}
+	req, err := http.NewRequest(http.MethodPost, usageStatsURL, bytes.NewBuffer(out))
+	if err != nil {
+		return err
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := httpClient.Do(req.WithContext(ctx))
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode/100 != 2 {
```

Comment:
It took me a few moments to parse what are we achieving here. Testing for the 2xx range until we know for sure what the server's response would be? In that case, can we somehow know right away?

Reply:
I copied this from Loki. We returning a `w.WriteHeader(http.StatusOK)` so I'm gonna change it here